File: bindings/csharp/speech_recognition_result_event_args.cs
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

namespace Carbon.Recognition.Speech
{
    /// <summary>
    /// Define payload of speech intermediate/final result events.
    /// </summary>
    public class SpeechRecognitionResultEventArgs : System.EventArgs
    {
        internal SpeechRecognitionResultEventArgs(Carbon.Internal.SpeechRecognitionEventArgs e)
        {
            this.Result = new SpeechRecognitionResult(e.Result);
            this.SessionId = e.SessionId;
        }

        /// <summary>
        /// Specifies the recognition result.
        /// </summary>
        public SpeechRecognitionResult Result { get; }

        /// <summary>
        /// Specifies the session identifier.
        /// </summary>
        public string SessionId { get; }

        /// <summary>
        /// Returns a string that represents the speech recognition result event.
        /// </summary>
        /// <returns>A string that represents the speech recognition result event.</returns>
        public override string ToString()
        {
            return string.Format("SessionId:{0} ResultId:{1} Status:{2} Recognized text:<{3}>.", SessionId, Result.ResultId, Result.Reason, Result.RecognizedText);
        }
    }
}

File: bindings/java/com/microsoft/cognitiveservices/speech/recognition/speech/SpeechRecognitionResultEventArgs.java
package com.microsoft.cognitiveservices.speech.recognition.speech;
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

/// <summary>
/// Define payload of speech intermediate/final result events.
/// </summary>
public class SpeechRecognitionResultEventArgs //: System.EventArgs
{
    SpeechRecognitionResultEventArgs(com.microsoft.cognitiveservices.speech.internal.SpeechRecognitionEventArgs e)
    {
        this._Result = new SpeechRecognitionResult(e.getResult());
        this._SessionId = e.getSessionId();
    }

    /// <summary>
    /// Specifies the recognition result.
    /// </summary>
    public SpeechRecognitionResult getResult()
    {
        return _Result;
    }// { get; }
    private SpeechRecognitionResult _Result;

    /// <summary>
    /// Specifies the session identifier.
    /// </summary>
    public final String getSessionId()
    {
        return _SessionId;
    }// { get; }
    private String _SessionId;

    /// <summary>
    /// Returns a String that represents the speech recognition result event.
    /// </summary>
    /// <returns>A String that represents the speech recognition result event.</returns>
    @Override
    public String toString()
    {
        return "SessionId:" + _SessionId + " ResultId:" + _Result.getResultId() + " Status:" + _Result.getReason() + " Recognized text:<" + _Result.getRecognizedText() + ">.";
    }
}
